h2d/Text.hx

```diff
@@ -236,11 +237,12 @@ class Text extends Drawable {
 					var e = font.getChar(cc);
 					size += e.width + letterSpacing + e.getKerningOffset(prevChar);
 					prevChar = cc;
-					if( font.charset.isBreakChar(cc) ) break;
+					var nc = text.charCodeAt(k+1);
+					if( font.charset.isBreakChar(cc) && (nc == null || !font.charset.isComplementChar(nc)) ) break;
 				}
 				if( size > maxWidth || (!breakFound && size + afterData > maxWidth) ) {
 					newline = true;
-					if( font.charset.isSpace(cc) ){
+					if( font.charset.isBreakChar(cc) && (nc == null || !font.charset.isComplementChar(nc)) ){

```

Comment:
It should not check for `isBreakChar`, and will introduce issues. It checks for space in order to trim whitespace after performing line break, since regular space have to be removed, but not renderable character (i.e. Chinese/Japanese characters). With your change, it will remove the characters it shouldn't, when there's no complement character present.

Reply:
Indeed. I pulled that from an older and modified heaps branch, it's irrelevant here.
I added complement chars to HtmlText too.

Thanks for review !